app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -110,7 +114,17 @@ class PreviewBadge : View {
             this.ifMarquee = currentState.getBoolean(BUNDLE_MARQUEE)
             this.badgeSpeed = currentState.getInt(BUNDLE_SPEED)
             this.badgeMode = Mode.values()[currentState.getInt(BUNDLE_MODE)]
+            this.countFrame = currentState.getInt(BUNDLE_CNTFRAME)
+            this.lastFrame = currentState.getInt(BUNDLE_LSTFRAME)
             this.checkList = currentState.getParcelableArrayList(BUNDLE_CHECKLIST)
+
+            valueAnimator?.cancel()
+
+            countFrame = 0
+            lastFrame = 0
```

Comment:
Why?

Reply:
I am resetting the valueanimator, because both the animators were controlling the animation

Reply:
CountFrame and Lastframe are used to calculate the change in frame, as we are calculating the frames from the divisor and as the number is large, the divisor is same most of the time, these two variables make us check only for the next new number

Reply:
So I am keeping track of the last count and the frame to start the animation from the exact frame

Reply:
You took them from the bundle and then reassigned to 0

Reply:
Omg, I'm so sorry

Reply:
Let me not store them anymore and restart the animation, that would be better